### PR TITLE
bugfix: add judge for whether pidfile path is given

### PR DESCRIPTION
**Pull request by Ace-Tang**
Signed-off-by: Ace-Tang <aceapril@126.com>

### Ⅰ. Describe what this PR did

save pid into pidfile only when pidfile path is given, make code more reasonable.

### CODE REVIEW

**Comment by HusterWan** [comment]
@Ace-Tang please create PR base on branch `0.5.x`, thanks a lot !!!

File: main.go [hunk 1]
```diff
@@ -154,15 +154,17 @@ func runDaemon() error {
...
 
 	// saves daemon pid to pidfile.
-	if err := utils.NewPidfile(cfg.Pidfile); err != nil {
-		logrus.Errorf("failed to create pidfile: %s", err)
-		return err
-	}
-	defer func() {
-		if err := os.Remove(cfg.Pidfile); err != nil {
-			logrus.Errorf("failed to delete pidfile: %s", err)
+	if cfg.Pidfile != "" {
```
Comment (HusterWan) [inline main.go]: If Pidfile exists, do we need to truncate it's content ?

Comment (Ace-Tang) [reply main.go]: Look at `utils.NewPidfile`, it won't go ahead when pidfile has value, it a common logic most process deal with.

**Comment by HusterWan** [comment]
LGTM

**Pull request merged**